yabt/buildcontext.py

```diff
@@ -417,7 +417,7 @@ def run_in_buildenv(
         docker_run.append(format_qualified_image_name(buildenv_target))
         docker_run.extend(cmd)
         logger.info('Running command in build env "{}" using command {}',
-                    buildenv_target_name, docker_run)
+                    buildenv_target_name, ' '.join(docker_run))
```

Comment:
this will not handle quoting and escaping parameters correctly.

probably most common cases could be handled with something like

```py
def format_for_cli(part):
    return '"{}"'.format(part) if ' ' in part else part

' '.join(format_for_cli(part) for part in docker_run)
```

would be nice to find a library that does that for you though :-)

Reply:
Can you give an example of a problematic `docker_run` list?

Reply:
> Can you think of any other problems besides the spaces?

it could be a rabbit hole. if it has a quote inside then the quote should be escaped. if it has `\` you'll need double escaping.

since these are usually pretty low-level commands I don't expect this will be a problem often, and it doesn't have to be perfect because it's intended for human to copy and paste, not for machine to take and run correctly (I hope!), so just handling spaces should be a good start.